extensions/extensions-configuration-store-vault/src/main/java/com/boozallen/aissemble/configuration/dao/VaultPropertyDao.java

```diff
@@ -0,0 +1,153 @@
+        if (property != null) {
+            try {
+                final Vault vault = getVault();
+                this.write(vault, Collections.singletonList(property));
+            } catch (VaultException e) {
+                throw new PropertyDaoException(String.format("Unable to write to vault with property: %s.", property), e);
+            }
+        } else {
+            throw new PropertyDaoException("Unable to write to vault with property: null");
+        }
+    }
+
+    /**
+     * Write given set of properties to the vault
+     * @param properties to be written to vault
+     * @throws PropertyDaoException
+     */
+    @Override
+    public void write(Set<Property> properties) throws PropertyDaoException {
+        if (properties != null && !properties.isEmpty()) {
+            try {
+                final Vault vault = getVault();
+                this.write(vault, new ArrayList<>(properties));
+            } catch (VaultException e) {
+                throw new PropertyDaoException(String.format("Unable to write to vault with properties: %s.", properties), e);
+            }
+        } else {
+            throw new PropertyDaoException("Unable to write to vault with empty properties");
+        }
+
+    }
+
+    private void write(Vault vault, List<Property> properties) throws VaultException {
+        for (Property property : properties) {
+            // path: properties/<<group name>>
+            String path = String.format("properties/%s", property.getGroupName());
+            // create a property secret engine
+            createPropertiesSecretEngine(vault, path);
+
+            Map<String, Object> data = new HashMap<>();
+            data.put(property.getName(), property.getValue());
+            logger.info(String.format("Write to vault with properties %s: %s", path, data));
+
+            // write key as a secret to Vault
+            vault.logical().write(String.format("%s/%s", path, property.getName()), data);
+        }
+    }
+
+
+    private Vault getVault() throws VaultException {
+        DataEncryptionConfiguration config = KrauseningConfigFactory.create(DataEncryptionConfiguration.class);
+        final VaultConfig vaultConfig = new VaultConfig()
+                .address(config.getSecretsHostUrl())
+                .token(config.getSecretsRootKey())
+                .prefixPathDepth(2) // properties/<group name>/
+                .engineVersion(2) // use version 2 to create properties as secret
+                .build();
+
+        final Vault vault = new Vault(vaultConfig);
+
+        VaultUtil.checkSealStatusAndUnsealIfNecessary(vault);
+
+        return vault;
+    }
+
+    private MountResponse createPropertiesSecretEngine(Vault vault, String path) throws VaultException {
+        MountResponse response = null;
+        try {
+            response = vault.mounts().read(path);
+        } catch (Exception e) {
+            logger.warn("Configuration properties secret engine may not exist on vault");
```

Comment:
Q/A: Would this always be expected on the first run, or is the expected case that it's already configured before the service is ever used?  If the former, we might downgrade this to `INFO`.